Assets/DanmakU/Runtime/Core/DanmakuPool.cs

```diff
@@ -192,6 +192,7 @@ public class DanmakuPool : IEnumerable<Danmaku>, IDisposable {
     for (var i = 0; i < count; i++) {
       Times[activeCount + i] = 0f;
       danmaku[i] = new Danmaku(this, activeCount + i);
+      OldPositions[danmaku[i].Id] = danmaku[i].Position;
```

Comment:
Please move this to the Danmaku contructor instead by setting OldPosition to Position's current value here. This should be a general property held by all Danmaku created.

Reply:
This code is already unsafe, (but unused, as far as I can tell). So I removed this line, since presumably the caller would need to initialize any Danmaku returned by it anyways.